### PR TITLE
Add support for additional editorconfig file

### LINKED ISSUE #210: Add support for editorconfig path
Hello,
it would be cool if we could configure the path of the `.editorconfig` file (using ktlint `--editorconfig` flag) within KtlintExtension.

### PR DESCRIPTION
**Pull request by Tapchicoma**
Closes #210

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
Can you specify only one or does it support multiple files?

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt [hunk 1]
```diff
@@ -37,12 +37,23 @@ internal inline fun <reified T : Task> Project.registerTask(
 
 internal const val EDITOR_CONFIG_FILE_NAME = ".editorconfig"
 
-internal fun getEditorConfigFiles(currentProject: Project): FileCollection {
-    return searchEditorConfigFiles(
+internal fun getEditorConfigFiles(
+    currentProject: Project,
+    additionalEditorconfigFile: RegularFileProperty
+): FileCollection {
+    var editorConfigFileCollection = searchEditorConfigFiles(
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: Should we even search if the user has specified their own? If there is both one in the repository and one passed via the command line, which one does ktlint use?

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: According to ktlint code - it is still using working directory `.editorconfig` file even if `--editorconfig` option was used:
- loading editorconfig passed via `--editorconfig` option:
https://github.com/pinterest/ktlint/blob/af7202536fd76cce9b3bb6ff0f81855de5060f3e/ktlint/src/main/kotlin/com/pinterest/ktlint/Main.kt#L379
- loading current working directory editorconfigs:
https://github.com/pinterest/ktlint/blob/af7202536fd76cce9b3bb6ff0f81855de5060f3e/ktlint/src/main/kotlin/com/pinterest/ktlint/Main.kt#L387

`editorConfigFileCollection` is just used as a task input, but never passed to the ktlint itself, so merging here all editorconfig files is right behaviour.

Though, actually, I've noticed that in this PR additional `.editorconfig` file was never passed to ktlint itself :+1:

**Comment by Tapchicoma** [comment]
> Can you specify only one or does it support multiple files?

Actually type of the extension field is `RegularFileProperty`, so only one file could be set.

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt (hunk 1 above)
Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: Fixed in https://github.com/JLLeitschuh/ktlint-gradle/pull/247/commits/eab563de0cfa9d561b32e00d77a7f427d463f903

Comment (JLLeitschuh) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: If it's an override, then it should be called an override, not `additional`. `additional` sounds like they are being merged or something.

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: Sorry, maybe I was not clear - ktlint adds additional `.editorconfig` file to the `.editorconfig` files that are detected in the working directory.

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt [hunk 2]
```diff
@@ -37,12 +37,23 @@ internal inline fun <reified T : Task> Project.registerTask(
...
         currentProject,
         currentProject.projectDir.toPath(),
         currentProject.files()
     )
+
+    if (additionalEditorconfigFile.isPresent) {
+        editorConfigFileCollection = editorConfigFileCollection.plus(
+            currentProject.files(additionalEditorconfigFile.asFile.get().toPath())
+        )
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: Ohhhh. I didn't spot this was an `add`. That's why I was so confused.

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt (hunk 1 above)
Comment (JLLeitschuh) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: I think I understand now. I didn't see that it's an `add` below. 🤦‍♂

File: plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/TestsCommon.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    //language=Groovy
+    buildFile().writeText(
+        """
+            $pluginsBlockWithMainPluginAndKotlinJvm
+
+            repositories {
+                gradlePluginPortal()
+            }
+
+            import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
```
Comment (JLLeitschuh) [inline plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/TestsCommon.kt]: 🤔 Didn't know you could have an import statement in the middle of a file in groovy scripts. Can you do the same thing in Kotlin?! Huh.

File: plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/EditorConfigTests.kt [hunk 4]
```diff
@@ -0,0 +1,167 @@
+package org.jlleitschuh.gradle.ktlint
```
Comment (JLLeitschuh) [inline plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/EditorConfigTests.kt]: Good idea extracting this to a separate test.

**Pull request merged**